Simplified/NYPLBookRegistry.m

```diff
@@ -802,7 +808,7 @@ - (NSArray *)booksMatchingStateMask:(NSUInteger)mask
      enumerateKeysAndObjectsUsingBlock:^(__attribute__((unused)) NSString *identifier,
                                          NYPLBookRegistryRecord *const record,
                                          __attribute__((unused)) BOOL *stop) {
-       if (record.state & mask) {
+       if (states.count == 0 || [states containsObject:@(record.state)]) {
```

Comment:
hmm I don't fully understand the reason for checking `states.count == 0`... if we have no states coming in we still add the book?

2nd thing: from the old code it seems like `record.state` could be nil, we should still continue to check for that to avoid a possible crash (I think... I'm saying that because NSArray::containsObject wants a non-optional in swift, although I know we are in objc here).

Reply:
I am using an empty array as a convention of all enum cases, so that we don't need to compare the `record.state` to every enum case

Reply:
Added statement to check `record.state`